### ISSUE TITLE
[Help Wanted] [MM-13325] Design a sample email plugin for Mattermost

### ISSUE THREAD

**Issue by esethna** [mattermost/mattermost-server#9957]
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-13325)

We are looking for help to design an email plugin for Mattermost in two phases.

PHASE 1: Send emails directly to a Mattermost channel

Minimum functionality:
 - Ability to connect your email account to Mattermost
 - Ability to send email notifications to a channel of your choice (either public, private or direct message channel)

PHASE 2: Respond to emails directly in a Mattermost channel

Minimum functionality:
 - Ability to respond to an email from within a Mattermost channel
 - Ability to add/remove who the response is sent to

For sample plugins, see https://github.com/mattermost/mattermost-plugins.

This is an open-ended design where you have the opportunity to add your fingerprint on the final look and feel of the plugin. If you're interested, let us know here or message @jason.blais at https://community.mattermost.com.

**Comment by bullsseye** [comment]
Hi. I want to pick this up.

**Comment by hanzei** [comment]
Thanks @bullsseye! Feel free to join the Developer Toolkit channel on https://community.mattermost.com for further dicussion about this and plugins in general. If you have any question or need an early review, let uns know.

**Comment by bullsseye** [comment]
In my understanding, every channel will be associated to an email id, when this email id will receive a message, I will post this message to the channel. 
In the reply case,
1. What will be the email ids I will reply to ?. If there are certain email addresses I will take from the user who is replying, should this be done at the channel level ?
2. When somebody will be sending a reply, the "from" email address in this case should be user's email address or channel's ?

**Comment by hanzei** [comment]
1. I don't know if I get your question right. Could you rephrase it?
2. I would say, the e-mail adress a channel is subscripted to should be taken.

**Comment by hanzei** [comment]
@bullsseye Just want to check if my feedback was clear enough. Do you have any other questions?
